lib/ansible/modules/cloud/amazon/lambda.py

```diff
@@ -350,29 +368,26 @@ def main():
 
         # If VPC configuration is desired
         if vpc_subnet_ids or vpc_security_group_ids:
-            if len(vpc_subnet_ids) < 1:
-                module.fail_json(msg='At least 1 subnet is required')
-
-            if len(vpc_security_group_ids) < 1:
-                module.fail_json(msg='At least 1 security group is required')
+            if vpc_subnet_ids is None or len(vpc_subnet_ids) == 0 or vpc_security_group_ids is None or len(vpc_security_group_ids) == 0:
+                module.fail_json(msg='vpc connectivity requires at least one security group and one subnet')
 
             if 'VpcConfig' in current_config:
                 # Compare VPC config with current config
                 current_vpc_subnet_ids = current_config['VpcConfig']['SubnetIds']
                 current_vpc_security_group_ids = current_config['VpcConfig']['SecurityGroupIds']
 
-                subnet_net_id_changed = sorted(vpc_subnet_ids) != sorted(current_vpc_subnet_ids)
-                vpc_security_group_ids_changed = sorted(vpc_security_group_ids) != sorted(current_vpc_security_group_ids)
+                subnets_changed = sorted(vpc_subnet_ids) != sorted(current_vpc_subnet_ids)
+                security_groups_changed = sorted(vpc_security_group_ids) != sorted(current_vpc_security_group_ids)
```

Comment:
ryansb left a comment on https://github.com/ansible/ansible/pull/26059 requesting that you don't change the var names. 
"I don't see the value in changing these var names - can you change them back to what they were to preserve authorship?"
"Still don't see these changes as outweighing the loss of authorship/blame."